src/script/standard.h

```diff
@@ -73,6 +73,22 @@ class CNoDestination {
     friend bool operator<(const CNoDestination &a, const CNoDestination &b) { return true; }
 };
 
+struct PKHash : public uint160
+{
+    PKHash() : uint160() {}
+    explicit PKHash(const uint160& hash) : uint160(hash) {}
+    explicit PKHash(const CPubKey& pubkey);
+    using uint160::uint160;
+};
+
+struct SHash : public uint160
```

Comment:
Concept ACK—I dislike the name `SHash` though, it sounds too general. I'd prefer a longer name that makes it more clear what is being hashed, maybe `ScriptHash`?

Reply:
done